dotCMS/src/integration-test/java/com/dotcms/publishing/job/SiteSearchJobImplTest.java

```diff
@@ -439,74 +445,220 @@ public void Test_Incremental_Job_Test_Pages_Are_Found_Create_And_Publish_New_Pag
+        boolean siteSearchDefaultLanguage;
+        boolean siteSearchSecondLanguage;
+        boolean siteSearchThirdLanguage;
+        boolean createContentInDefaultLanguage;
+        boolean createContentInSecondLanguage;
+        boolean createContentInThirdLanguage;
+        boolean createPageInDefaultLanguage;
+        boolean createPageInSecondLanguage;
+        boolean expectedResultsWhenSearchingContentInDefaultLanguage;
+        boolean expectedResultsWhenSearchingContentInSecondLanguage;
+        boolean expectedResultsWhenSearchingContentInThirdLanguage;
+
+        public TestCaseSiteSearch(boolean defaultPageToDefaultLanguage,
+                boolean defaultContentToDefaultLanguage, boolean siteSearchDefaultLanguage,
+                boolean siteSearchSecondLanguage, boolean siteSearchThirdLanguage,
+                boolean createContentInDefaultLanguage,
+                boolean createContentInSecondLanguage, boolean createContentInThirdLanguage,
+                boolean createPageInDefaultLanguage,
+                boolean createPageInSecondLanguage,
+                boolean expectedResultsWhenSearchingContentInDefaultLanguage,
+                boolean expectedResultsWhenSearchingContentInSecondLanguage,
+                boolean expectedResultsWhenSearchingContentInThirdLanguage) {
+            this.defaultPageToDefaultLanguage = defaultPageToDefaultLanguage;
+            this.defaultContentToDefaultLanguage = defaultContentToDefaultLanguage;
+            this.siteSearchDefaultLanguage = siteSearchDefaultLanguage;
+            this.siteSearchSecondLanguage = siteSearchSecondLanguage;
+            this.siteSearchThirdLanguage = siteSearchThirdLanguage;
+            this.createContentInDefaultLanguage = createContentInDefaultLanguage;
+            this.createContentInSecondLanguage = createContentInSecondLanguage;
+            this.createContentInThirdLanguage = createContentInThirdLanguage;
+            this.createPageInDefaultLanguage = createPageInDefaultLanguage;
+            this.createPageInSecondLanguage = createPageInSecondLanguage;
+            this.expectedResultsWhenSearchingContentInDefaultLanguage = expectedResultsWhenSearchingContentInDefaultLanguage;
+            this.expectedResultsWhenSearchingContentInSecondLanguage = expectedResultsWhenSearchingContentInSecondLanguage;
+            this.expectedResultsWhenSearchingContentInThirdLanguage = expectedResultsWhenSearchingContentInThirdLanguage;
+        }
+    }
+
+    @DataProvider
+    public static Object[] siteSearchTestCases() {
+
+        /*
+         * Given sceneario: default-language content referenced from a page in default lang only. DEFAULT_PAGE_TO_DEFAULT_LANGUAGE=true.
+         * Create a Site-Search run only in default language
+         * Expected: searching for the content in default language should give results. The result should be the page in the default language
+         */
+
+        TestCaseSiteSearch case1 = new TestCaseSiteSearchBuilder()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'case1' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5303548)